src/hammr/hammr.py

```diff
@@ -173,63 +164,31 @@ def check_credfile(credfile):
     mainParser.print_help()
     exit(0)
 
-if mainArgs.url is not None:
-    url=mainArgs.url
-
-if mainArgs.user is not None:
-    if not mainArgs.password:
-        mainArgs.password = getpass.getpass()
-    username=mainArgs.user
-    password=mainArgs.password
-    sslAutosigned=True
+credentials = credentials_utils.Credential()
+if mainArgs.user is not None and mainArgs.publicKey is not None and mainArgs.secretKey is not None:
+	#using API key cmd lines
+	credentials.fill_credentials_from_cmd_apiKey(mainArgs)
```

Comment:
Don't use mainArgs in parameters but just fill your Credential structure before calling this method without parameters.
Please check for all method in Credential using mainArgs.